protostar/cli/argument_value_from_config_extractor.py

```diff
@@ -0,0 +1,65 @@
+from typing import Any, Optional, Protocol
+
+
+class ArgumentValueFromConfigProviderProtocol(Protocol):
+    def get_argument_value(
+        self, command_name: str, argument_name: str, profile_name: Optional[str] = None
+    ) -> Optional[Any]:
+        ...
+
+    def get_shared_argument_value(
+        self, argument_name: str, profile_name: Optional[str] = None
+    ) -> Optional[Any]:
+        ...
+
+
+class ArgumentValueFromConfigExtractor:
```

Comment:
What is the purpose of this class? The name does not suggest it at all

Reply:
Choosing correct argument (shared arguments, command specific argument, profiled shared argument, profiled command specific command).

`ArgumentValueFromConfigStrategy`?
@Arcticae @MaksymilianDemitraszek Suggestions?

Reply:
This class it basically one method, I would merge it with ` ArgumentValueFromConfigProviderProtocol`. As we have only one strategy, there is no point in splitting it. (We would avoid this weird composition this way and code would be simpler)

Reply:
Merged. Closing this thread in favor the one below.